src/kudu/consensus/leader_election.cc

```diff
@@ -507,28 +518,12 @@ FlexibleVoteCounter::IsPessimisticQuorumSatisfied() const {
   VLOG_WITH_PREFIX(3) << "Checking if pessimistic quorum is satisfied.";
 
   // Fetching all regions.
-  std::vector<std::string> regions;
+  std::set<std::string> regions;
   for (const std::pair<std::string, int>& region_count_pair :
       voter_distribution_) {
-    regions.push_back(region_count_pair.first);
-  }
-
-  // Getting quorum satisfaction for individual regions.
-  std::vector<std::pair<bool, bool> > results =
-      IsMajoritySatisfiedInRegions(regions);
-
-  CHECK_EQ(results.size(), regions.size());
-
-  // Computing the final value.
-  bool quorum_satisfied = true;
-  bool quorum_satisfaction_possible = true;
-
-  for (const std::pair<bool, bool>& result : results) {
-    quorum_satisfied = quorum_satisfied && result.first;
-    quorum_satisfaction_possible =
-        quorum_satisfaction_possible && result.second;
+    regions.insert(region_count_pair.first);
   }
-  return std::make_pair<>(quorum_satisfied, quorum_satisfaction_possible);
+  return IsMajoritySatisfiedInRegions(regions);
```

Comment:
I would argue that the previous code organization is more intuitive.
The helper function IsMajoritySatisfiedInRegions (in new code) which takes a set of 
regions and returns a single pair only makes sense when we are in Pessimistic Quorum
mode. In other modes like majority of majorities mode, we can't do simple && enforcement
on the quorum satisfaction results for each region. Therefore having it inlined in IsPessimisticQuorumSatisfied was causing less confusion.

If you want to still break it up into 2 functions for readability reasons, I would say that the
function  IsMajoritySatisfiedInRegions should be renamed as IsMajoritySatisfiedInRegionsConjunctive or IsMajoritySatisfiedInAllRegions ... something that conveys && of all regions.

Reply:
Will change it to IsMajoritySatisfiedInAllRegions()